lib/redlock/client.rb

```diff
@@ -262,17 +273,29 @@ def lock_instances(resource, ttl, options)
       value = (options[:extend] || { value: SecureRandom.uuid })[:value]
       allow_new_lock = options[:extend_only_if_locked] ? 'no' : 'yes'
 
+      errors = []
+
       locked, time_elapsed = timed do
-        @servers.select { |s| s.lock resource, value, ttl, allow_new_lock }.size
-      end
+        @servers.select do |s|
+          s.lock(resource, value, ttl, allow_new_lock)
+        rescue RedisClient::ConnectionError => e
```

Comment:
I think any exception should be counted as an error, not only connection errors 🤔